packages/eslint-plugin/tests/lib/rules/react-require-autocomplete.test.js

```diff
@@ -0,0 +1,62 @@
+const {RuleTester} = require('eslint');
+
+const rule = require('../../../lib/rules/react-require-autocomplete');
+
+const ruleTester = new RuleTester({parser: require.resolve('babel-eslint')});
+
+function errorWithTagName(tagName) {
+  return [
+    {
+      type: 'JSXOpeningElement',
+      message: `Input elements of type '${
+        tagName || 'text'
+      }' must have an autoComplete attribute`,
+    },
+  ];
+}
+
+ruleTester.run('react-require-autocomplete', rule, {
+  valid: [
+    {
+      code: '<TextField type="email" autoComplete="email" />',
+    },
+    {
+      code: '<input type="email" autoComplete="email" />',
+    },
+    {
+      code:
+        'const autoComplete = "email"; const myInput = <input type="email" autoComplete={autoComplete} />;',
+    },
+    {
+      code: `
+      function MyComponent() {
+        const fieldProps = {
+          label: "Email",
+          autoComplete: "emaill"
+        }
+        return <TextField type="email" {...fieldProps} />;
+      }`,
+    },
+    {
+      code: `
+      function MyComponent({fieldProps}) {
+        return <TextField type="text" {...fieldProps} />;
+      }`,
+    },
+  ],
```

Comment:
I think we could add two more test cases:
-  for an `input` type on which `autoComplete` is not allowed (e.g. `<input type="checkbox" />`)
- a test where it doesn’t warn about components other than `input` and `TextField`